src/System.Device.Gpio/System/Device/Gpio/Drivers/RaspberryPi3Driver.Linux.cs

```diff
@@ -472,6 +512,21 @@ private void Initialize()
 
                 Interop.close(fileDescriptor);
                 _registerViewPointer = (RegisterView*)mapPointer;
+
+                // Detect whether we're running on a Raspberry Pi 4
+                IsPi4 = false;
+                try
+                {
+                    string model = File.ReadAllText("/proc/device-tree/model", Text.Encoding.ASCII);
+                    if (model.Contains("Raspberry Pi 4"))
+                    {
+                        IsPi4 = true;
+                    }
+                }
+                catch
```

Comment:
what exceptions do we expect? Can we prevent them with i.e. File.Exists?

Reply:
Maybe we can (File.Exists + permission check maybe) but I wouldn't know how to test the behavior on all possible other systems.

Reply:
Let's do following:
- guard against anything we know might go wrong (at minimum File.Exists)
- leave try catch but add Debug.Assert in there so it at least fails in debug (make sure to add a comment)